build/cmake/MongoPlatform.cmake

```
@@ -0,0 +1,72 @@
+#[[
+
+Defines a target mongo::detail::c_platform (alias of _mongo-platform), which
+exposes system-level supporting compile and link usage requirements. All targets
+should link to this target with level PUBLIC.
+
+Use mongo_platform_compile_options and mongo_platform_link_options to add usage
+requirements to this library.
+
+The mongo::detail::c_platform library is installed and exported with the
+bson-targets export set as an implementation detail. It is installed with this
+export set so that it is available to both libbson and libmongoc (attempting to
+install this target in both bson-targets and mongoc-targets export sets would
+lead to duplicate definitions of mongo::detail::c_platform for downstream
+users).
+
+]]
+
+add_library(_mongo-platform INTERFACE)
+add_library(mongo::detail::c_platform ALIAS _mongo-platform)
+set_property(TARGET _mongo-platform PROPERTY EXPORT_NAME detail::c_platform)
+install(TARGETS _mongo-platform EXPORT bson-targets)
```

```suggestion
install(TARGETS _mongo-platform EXPORT bson-targets)
install(TARGETS _mongo-platform EXPORT mongoc-targets)
```
Do we want to include the imported target in the mongoc-targets file as well? Or are we fine with obtaining it transitively via libbson?

---

Seems a little out-of-place. Consider moving the `install()` command into `src/libbson/CMakeLists.txt`?

---

It has to be exported only once, otherwise it will be defined twice and goofs will ensue. I chose the bson-targets export set only because it is pulled by both bson and mongoc. An "ideal" would be to define a third export set, but I think that has diminishing value when it can just ride along with the bson targets. I've also tucked it into the "detail" namespace so that we "reserve the right" to move/break/change/delete it in the future.